### PR TITLE
feat: MUSD-126 Add mUSD CTA to asset details page

### PR DESCRIPTION
**Pull request by Matt561**
## **Description**
Adds the mUSD conversion CTA to the asset details page for supported conversion tokens. In the event that the asset is supported in the mUSD conversion flow **and** stablecoin lending we favor the mUSD conversion CTA. 

Read Cursor summary below for detailed changes.

## **Changelog**

CHANGELOG entry: added mUSD CTA to asset overview screen

## **Related issues**

Fixes: [MUSD-126: Add a convert to mUSD CTA card to the asset details page for supported conversion stablecoins](https://consensyssoftware.atlassian.net/browse/MUSD-126)

## **Manual testing steps**

```gherkin
Feature: mUSD Conversion CTA on Asset Overview

  Scenario: user views mUSD conversion CTA for eligible stablecoins
    Given user has mainnet USDC, USDT, or DAI in their wallet
    And mUSD conversion feature flag is enabled

    When user navigates to the token details page for the eligible stablecoin
    Then user sees "Earn rewards when you convert to mUSD" CTA
    And the CTA displays the mUSD icon

  Scenario: user initiates mUSD conversion from asset overview
    Given user is on the asset overview page for USDC
    And user has not seen the mUSD education screen

    When user taps "mUSD" link in the CTA
    Then user is navigated to the mUSD education screen

  Scenario: returning user initiates mUSD conversion
    Given user has previously seen the mUSD education screen
    And user is on the asset overview page for USDC

    When user taps "mUSD" link in the CTA
    Then user is navigated directly to the mUSD conversion confirmation screen
```

### **Before**

N/A - CTA didn't exist
### **After**

https://github.com/user-attachments/assets/9318e291-d8df-47d9-8df7-c2d53c35bf82
## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

---

> [!NOTE]
> Adds mUSD conversion CTAs on asset overview and token list, gated by feature flag and integrated with conversion/education flow; updates styles, tests, and i18n.
> 
> - **Earn (mUSD Conversion)**:
>   - Add `MusdConversionAssetOverviewCta` (+ styles/tests) and render it in `EarnLendingBalance` via `renderCta`, favoring it over the lending empty state when enabled and token is convertible.
>   - Add `MusdConversionAssetListCta` (+ styles/tests) and display it in `UI/Tokens` when the feature flag is on.
>   - Wire up `useMusdConversionTokens`, `useMusdConversion`, and `selectIsMusdConversionFlowEnabledFlag`; add new test IDs in `EARN_TEST_IDS`.
>   - Update `StakeButton` to handle convertible stablecoins (initiate conversion/education flow) and refactor button label logic.
> - **Styles/UI tweaks**:
>   - `TokenDetails.styles.tsx`: reduce `tokenDetailsContainer` `marginTop` to `16`.
>   - `EarnLendingBalance`: layout adjustments (button container, `paddingTop`, larger `AvatarToken`), add spacing for CTAs.
>   - `StakingCta`: center layout/text.
> - **Tests**:
>   - Comprehensive new/updated tests and snapshots across Earn, Tokens, Asset Overview, and confirmation views to cover new CTAs, feature flag behavior, and navigation.
> - **i18n**:
>   - Add strings: `earn.musd_conversion.earn_rewards_when`, `earn.musd_conversion.you_convert_to`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit afc5018acc02a8b55a7e8f589d586e5bbac607c8. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Comment by Matt561** [comment]
Skipping Sonarcloud since missing coverage is on existing code unrelated to this PR and will be addressed in a separate PR.

Edit: Disregard, looks like coverage is okay now.

File: app/components/UI/Tokens/index.tsx [hunk 1]
```diff
@@ -225,7 +225,11 @@ const Tokens = memo(({ isFullView = false }: TokensProps) => {
         </Box>
       ) : sortedTokenKeys.length > 0 ? (
         <>
-          {isMusdConversionFlowEnabled && <MusdConversionCta />}
+          {isMusdConversionFlowEnabled && (
+            <View style={isFullView ? tw`px-4` : undefined}>
+              <MusdConversionAssetListCta />
+            </View>
+          )}
```
Comment (Matt561) [inline app/components/UI/Tokens/index.tsx:232]: This is to ensure consistent padding in full view.

**Review by Prithpal-Sooriya — APPROVED** [review]
Asset CO Files LGTM

**Review by matthewwalsh0 — APPROVED** [review]

**Review by nickewansmith — APPROVED** [review]
lgtm
